### PR TITLE
feat: add bookings [part 1]: add establishments and addresses

### PR DESCRIPTION
**Pull request by thelastandrew**

### CODE REVIEW

File: src/components/bookings/AddAddressStep/BookingsAddAddressStep.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    setCurrentStep(AddAddressSteps.WorkingHours);
+  };
+
+  return (
+    <form onSubmit={onSubmit(handleSubmit)}>
+      {!currentEstablishment && (
+        <TextInput
+          withAsterisk
+          label={t('establishments.name')}
+          placeholder={t('establishments.estName').toString()}
```
Comment (remadyt) [inline src/components/bookings/AddAddressStep/BookingsAddAddressStep.tsx:57]: as string

File: src/components/bookings/AddAddressStep/BookingsAddAddressStep.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          mb="sm"
+          {...getInputProps('name')}
+        />
+      )}
+
+      <TextInput
+        withAsterisk
+        label={t('establishments.city')}
+        placeholder={t('establishments.cityPlaceholder').toString()}
```
Comment (remadyt) [inline src/components/bookings/AddAddressStep/BookingsAddAddressStep.tsx:66]: as string here and such places

File: src/components/bookings/AddWorkingHoursStep/BookingsAddWorkingHoursStep.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  handleOnClose,
+}) => {
+  const {
+    establishmentActions: { addEstablishmentAndAddress },
+    addressActions: { addAddress },
+  } = useBookings((state) => state);
+  const { t } = useTranslation();
+  const { classes } = useStyles();
+
+  const initialValues: any = {};
```
Comment (remadyt) [inline src/components/bookings/AddWorkingHoursStep/BookingsAddWorkingHoursStep.tsx:40]: fix any

File: src/components/bookings/Establishment/BookingsEstablishment.tsx [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    }
+  }, [currentEstablishment]);
+
+  const handleClick = (id: string): void => {
+    navigate(`${Paths.bookings}/${id}`);
+  };
+
+  const handleUpdateEstablishment = async (): Promise<void> => {
+    if (name) {
+      await updateEstablishment({ name });
```
Comment (remadyt) [inline src/components/bookings/Establishment/BookingsEstablishment.tsx:93]: remove async await here, in all places ( in components )

File: src/components/bookings/Establishment/BookingsEstablishment.tsx [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+      children: <Text size="sm">{t('establishments.agreeToDeleteEstablishment')}</Text>,
+      labels: { confirm: t('modals.btnDelete'), cancel: t('modals.btnCancel') },
+      confirmProps: { color: 'red' },
+      onConfirm: async () => {
+        await deleteEstablishment();
+      },
+    });
+  };
+
+  const addressCards = addresses
```
Comment (remadyt) [inline src/components/bookings/Establishment/BookingsEstablishment.tsx:111]: addresses &&

remove null

File: src/components/bookings/WorkingDaysInputs/BookingsWorkingDaysInputs.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    opacity: '0.6',
+    textDecoration: 'underline dotted',
+    textUnderlineOffset: '0.15em',
+  },
+});
+
+type Props = {
+  getInputProps: any;
+  handleApplyToAll: () => void;
+  values: any;
```
Comment (remadyt) [inline src/components/bookings/WorkingDaysInputs/BookingsWorkingDaysInputs.tsx:30]: fix any

File: src/components/bookings/WorkingDaysInputs/BookingsWorkingDaysInputs.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  values,
+}) => {
+  const { t } = useTranslation();
+  const { classes } = useStyles();
+
+  return (
+    <>
+      {weekdays.map((day) => (
+        <Container key={day} px={0}>
+          <Flex justify="space-between" mb={day === WeekDays.Mon ? 'xs' : 'sm'}>
```
Comment (remadyt) [inline src/components/bookings/WorkingDaysInputs/BookingsWorkingDaysInputs.tsx:45]: use sm only

File: src/constants/workingHours.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
+export const workingHours = [
```
Comment (remadyt) [inline src/constants/workingHours.ts:1]: WORKING_HOURS

why u do not use range datepicker?

File: src/main.css [hunk 9]
```diff
@@ -6,4 +6,5 @@
 
 .container {
   flex: 1 0 auto;
+  height: 1px;
```
Comment (remadyt) [inline src/main.css:9]: why need this

File: src/pages/Bookings/Bookings.tsx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  );
+  const [formValues, setFormValues] = useState<IFormValues>(initialValues);
+  const [opened, { open, close }] = useDisclosure(false);
+  const { currentEstablishment, getEstablishments } = useBookings(
+    (state) => state.establishmentActions,
+  );
+  const { t } = useTranslation();
+
+  const init = useCallback(async () => {
+    await getEstablishments();
```
Comment (remadyt) [inline src/pages/Bookings/Bookings.tsx:25]: remove async await

File: src/store/slices/bookings/addressSlice.ts [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+          errorHandler(error as Error);
+        } finally {
+          set(
+            produce((state: BookingStore) => {
+              state.addressActions.isAddressFetching = false;
+            }),
+          );
+        }
+
+        return;
```
Comment (remadyt) [inline src/store/slices/bookings/addressSlice.ts:68]: why need return here?

File: src/store/slices/bookings/addressSlice.ts [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+              state.addressActions.isAddressFetching = false;
+            }),
+          );
+        }
+      }
+    },
+
+    deleteAddress: async () => {
+      const { currentEstablishment } = get().establishmentActions;
+      const { currentAddress } = get().addressActions;
```
Comment (remadyt) [inline src/store/slices/bookings/addressSlice.ts:170]: ```const { addressActions: { currentAddress }, establishmentActions: { currentEstablishment } } = get()```

File: src/store/slices/bookings/addressSlice.ts [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+
+                orders.forEach(async (order) => {
+                  await deleteFirestoreData(DatabasePaths.Orders, order.id);
+                });
+
+                // set(produce((state: BookingStore) => { state.orderSlice.orders = []; state.orderSlice.currentOrder = null; }));
+              }
+            });
+
+            // set(produce((state: BookingStore) => { state.tableSlice.tables = []; state.currentTable = null; }));
```
Comment (remadyt) [inline src/store/slices/bookings/addressSlice.ts:224]: remove commented code

File: src/store/slices/bookings/establishmentSlice.ts [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+        set(
+          produce((state: BookingStore) => {
+            state.establishmentActions.isEstablishmentFetching = false;
+          }),
+        );
+      }
+    },
+
+    updateEstablishment: async (data) => {
+      const { currentEstablishment } = get().establishmentActions;
```
Comment (remadyt) [inline src/store/slices/bookings/establishmentSlice.ts:143]: cosnt { ... } = get();

File: src/store/slices/bookings/establishmentSlice.ts [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+            produce((state: BookingStore) => {
+              state.establishmentActions.isUpdating = false;
+            }),
+          );
+        }
+      }
+    },
+
+    deleteEstablishment: async () => {
+      const { currentEstablishment } = get().establishmentActions;
```
Comment (remadyt) [inline src/store/slices/bookings/establishmentSlice.ts:176]: and here

File: src/store/slices/bookings/types.ts [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+  Sun = 'Sun',
+}
+
+interface IDayHours {
+  isAvailable: boolean;
+  start: string;
+  finish: string;
+}
+
+export interface IWorkingHours {
```
Comment (remadyt) [inline src/store/slices/bookings/types.ts:36]: Record<WeekDays.Mon, IDayHours>

**Review by remadyt — CHANGES_REQUESTED** [review]

File: src/components/index.ts [hunk 17]
```diff
@@ -4,3 +4,9 @@ export * from 'components/layout/Header/LanguageSwitcher/HeaderLanguageSwitcher'
 export * from 'components/layout/Header/HeaderThemeToggler/HeaderThemeToggler';
 export * from 'components/layout/Header/HeaderUserMenu/HeaderUserMenu';
 export * from 'components/clients';
+export * from 'components/bookings/Establishment/BookingsEstablishment';
+export * from 'components/bookings/Address/BookingsAddress';
+export * from 'components/bookings/AddAddressStep/BookingsAddAddressStep';
+export * from 'components/bookings/AddWorkingHoursStep/BookingsAddWorkingHoursStep';
+export * from 'components/bookings/AddressSettings/BookingsAddressSettings';
+export * from 'components/bookings/WorkingDaysInputs/BookingsWorkingDaysInputs';
```
Comment (pyropetrick) [inline src/components/index.ts:12]: export * from 'components/bookings/WorkingDaysInputs/BookingsWorkingDaysInputs';
export * from 'components/bookings/AddWorkingHoursStep/BookingsAddWorkingHoursStep';
export * from 'components/bookings/Establishment/BookingsEstablishment';
export * from 'components/bookings/AddressSettings/BookingsAddressSettings';
export * from 'components/bookings/Address/BookingsAddress';
export * from 'components/bookings/AddAddressStep/BookingsAddAddressStep';

File: src/store/slices/bookings/types.ts (hunk 16 above)
Comment (pyropetrick) [reply src/store/slices/bookings/types.ts:36]: Record<WeekDays, IDayHours>

**Review by pyropetrick — CHANGES_REQUESTED** [review]

File: src/store/slices/bookings/addressSlice.ts [hunk 18]
```diff
@@ -0,0 +1,236 @@
...
+
+export interface IAdressActions {
+  addressActions: {
+    isAddressFetching: boolean;
+    addresses: IAddress[];
+    currentAddress: Nullable<IAddress>;
+
+    setCurrentAddress: (id: string) => Promise<void>;
+    addAddress: (data: Omit<IAddress, 'id' | 'tables'>) => Promise<void>;
+    updateAddress: (data: Omit<IAddress, 'id' | 'tables'>) => Promise<void>;
```
Comment (remadyt) [inline src/store/slices/bookings/addressSlice.ts:25]: create interface for data

File: src/store/slices/auth/signInSlice.ts [hunk 19]
```diff
@@ -89,12 +90,12 @@ export const signInSlice: GenericStateCreator<BoundStore> = (set, get) => ({
...
         if (user) {
           const userId = `user-${user.uid}`;
-          const isUserExist = await getFireStoreDataByFieldName(
+          const userData = await getFireStoreDataByFieldName<IUser>(
             DatabasePaths.Users,
             userId,
           );
 
-          if (isUserExist) {
+          if (userData) {
```
Comment (remadyt) [inline src/store/slices/auth/signInSlice.ts:98]: why rename?

File: src/store/slices/bookings/establishmentSlice.ts [hunk 20]
```diff
@@ -0,0 +1,249 @@
...
+          state.addressActions.currentAddress = null;
+        }),
+      );
+
+      try {
+        const { data } = await getFirestoreData<IEstablishment, string>(
+          DatabasePaths.Establishments,
+        );
+
+        if (data.length) {
```
Comment (remadyt) [inline src/store/slices/bookings/establishmentSlice.ts:56]:
```
if (!data.length) {
    set(
          produce((state: BookingStore) => {
            state.establishmentActions.establishments = [];
          }),
        );
   return;
}

// some code..
```

File: src/store/slices/tournaments/tournamentsSlice.ts [hunk 21]
```diff
@@ -180,13 +185,18 @@ export const tournamentsSlice: GenericStateCreator<BoundStore> = (set, get) => (
...
-      set(
-        produce((state: BoundStore) => {
-          state.currentTournament = data;
-        }),
+      const data = await getFireStoreDataByFieldName<ITournamentData>(
+        DatabasePaths.Tournaments,
+        id,
       );
+
+      if (data) {
```
Comment (remadyt) [inline src/store/slices/tournaments/tournamentsSlice.ts:193]: there are cases when the date may not be?

File: src/store/slices/clients/clientSlice.ts [hunk 22]
```diff
@@ -100,17 +101,19 @@ export const clientSlice: GenericStateCreator<BoundStore> = (set, get) => ({
...
         nickname,
         'nickName',
       );
 
-      set(
-        produce((state: BoundStore) => {
-          state.client = data;
-        }),
-      );
+      if (data) {
```
Comment (remadyt) [inline src/store/slices/clients/clientSlice.ts:110]: there are cases when the date may not be?

**Review by remadyt — CHANGES_REQUESTED** [review]

**Review by remadyt — APPROVED** [review]

**Review by pyropetrick — APPROVED** [review]

**Pull request merged**